### ISSUE TITLE
translation ideas and problems

### ISSUE THREAD

**Issue by ekg** [vgteam/vg#1557]
Idea: we should be able to invert translations (going from the old graph to the new one) such as to lift over alignments, annotations, or other independent objects that weren't touched in the augmentation process. This can be done by reversing the from: and to: paths in the translation object. It could also be added as a feature of the translator constructor. So far we didn't need this but it could be useful.

Bug? This translation object is somewhat weird. It includes a mapping with only a position. Is this needed by the translation semantics or is it a bug we should resolve?

```json
{
  "from": {
    "mapping": [
      {
        "position": {
          "node_id": 126,
          "offset": 3
        },
        "edit": [
          {
            "from_length": 1,
            "to_length": 1,
            "sequence": "G"
          }
        ]
      },
      {
        "position": {
          "node_id": 126,
          "offset": 4
        }
      }
    ]
  },
  "to": {
    "mapping": [
      {
        "position": {
          "node_id": 219
        },
        "edit": [
          {
            "from_length": 1,
            "to_length": 1
          }
        ]
      }
    ]
  }
}
```

How I made the above structure

```
vg construct -r small/x.fa -v small/x.vcf.gz >s.vg
vg index -x s.xg -k 16 s.gcsa s.vg
vg map -d s -s GGACAGCTCTCCCTTGGGTCCAGAGTGTATACGAATTAACTCTGTTCTGG >s.gam
vg mod -i s.gam -Z s.trans s.vg >/dev/null
```

**Comment by 6br** [comment]
It seems interesting idea. Using transitions, we can describe and lift over the differences between personal genomes/ alignments/annotations and existing variation graph.

In addition to this idea, for practical reasons or using as a database, transitions should be able to be annotated with the biological meanings. Can it be described in RDF? Is it related to #1555?

**Comment by JervenBolleman** [comment]
@6br yes it can be described in RDF. It's definitely related to #1555, as the easiest way to describe variants is to describe the difference between the variant and an reference.

It's possible but a pain to work with "string" edits in RDF I always suggest to have fully embedded paths and nodes.  Which kind of leads to a model where nodes can be paths by them selves. This sounds bizare at first utterance but I think works cleanly in the RDF world even though it is probably unusable in the VG internal datamodels.

I will ponder some more about it and put an example in this or the related issues.
